src/luxonis_ml/embeddings/emb_representative.py

```diff
@@ -0,0 +1,170 @@
+"""
+Find Representative Images from Embeddings
+
+This module offers techniques to identify representative images or embeddings within a dataset.
+This aids in achieving a condensed yet expressive view of your data.
+
+Methods:
+- Greedy Search: Aims to find a diverse subset of images by maximizing the minimum similarity to any image outside the set.
+  
+- K-Medoids: An adaptation of the k-means clustering algorithm, it partitions data into k clusters, each associated with a medoid.
+
+Main Applications:
+- Dataset Reduction: Helps in representing large datasets with a minimal subset while retaining the essence.
+  
+- Validation Set Creation: Identifies diverse samples for a robust validation set.
+
+Dependencies:
+- numpy
+- scikit-learn
+- kmedoids
+- luxonis_ml
+
+Example Usage:
+1. Greedy Search:
+    ```python
+    # Assuming you have 'embeddings' as a numpy array of shape (num_images, embedding_dim)
+    similarity_matrix = calculate_similarity_matrix(embeddings)
+    desired_size = int(len(embeddings) * 0.1)
+    selected_image_indices = find_representative_greedy(1-similarity_matrix, desired_size)
+    ```
+
+2. K-Medoids:
+    ```python
+    # to get all embeddings from qdrant:
+    ids, embeddings = get_all_embeddings(qdrant_client, collection_name="mnist")
+    # Assuming you have 'embeddings' as a numpy array of shape (num_images, embedding_dim)
+    similarity_matrix = calculate_similarity_matrix(embeddings)
+    desired_size = int(len(embeddings) * 0.1)
+    selected_image_indices = find_representative_kmedoids(similarity_matrix, desired_size)
+    ```
+
+"""
+
+from sklearn.metrics.pairwise import cosine_similarity
+from kmedoids import KMedoids
+
+from luxonis_ml.embeddings.qdrant_utils import *
+
+
+def calculate_similarity_matrix(embeddings):
+    return cosine_similarity(embeddings)
```

Comment:
Do we need this wrapper?

Reply:
There was an idea behind it. Firstly the user can define his own function for calculating the similarity matrix (like maybe based on some other distance) and also so the user can calculate similarity without separately importing sklearn.